spec/models/info_request_spec.rb

```diff
@@ -1066,79 +1067,54 @@
+        with(:all, hash_including(
+          {:conditions => include(/prominence != 'backpage'/)}))
       InfoRequest.find_old_unclassified({:conditions => ["prominence != 'backpage'"]})
     end
 
-    it 'should ask the database for requests that are awaiting description, have a last public response older
-        than 21 days old, have a user, are not the holding pen and are not backpaged' do
-      expected_conditions = ["awaiting_description = ?
-                                    AND (SELECT info_request_events.created_at
-                                         FROM info_request_events, incoming_messages
-                                         WHERE info_request_events.info_request_id = info_requests.id
-                                         AND info_request_events.event_type = 'response'
-                                         AND incoming_messages.id = info_request_events.incoming_message_id
-                                         AND incoming_messages.prominence = 'normal'
-                                         ORDER BY created_at desc LIMIT 1) < ?
-                                    AND url_title != 'holding_pen'
-                                    AND user_id IS NOT NULL".split(' ').join(' '),
-                             true, Time.now - 21.days]
-      expected_select = "*, (SELECT info_request_events.created_at
-                                   FROM info_request_events, incoming_messages
-                                   WHERE info_request_events.info_request_id = info_requests.id
-                                   AND info_request_events.event_type = 'response'
-                                   AND incoming_messages.id = info_request_events.incoming_message_id
-                                   AND incoming_messages.prominence = 'normal'
-                                   ORDER BY created_at desc LIMIT 1)
-                                   AS last_response_time".split(' ').join(' ')
+    it 'asks the database for requests that are awaiting description, have a
+        last public response older than 21 days old, have a user, are not the
+        holding pen' do
       expect(InfoRequest).to receive(:find) do |all, query_params|
-        query_string = query_params[:conditions][0]
-        query_params[:conditions][0] = query_string.split(' ').join(' ')
-        expect(query_params[:conditions]).to eq(expected_conditions)
-        expect(query_params[:select].split(' ').join(' ')).to eq(expected_select)
-        expect(query_params[:order]).to eq("last_response_time")
+        # check for awaiting description
+        expect(query_params[:conditions][0]).
+          to include("awaiting_description = ?")
+        expect(query_params[:conditions][1]).to eq(true)
+
+        # check last public response older than 21 days
+        expect(query_params[:conditions][0]).
+          to include("last_public_response_at < ?")
+        expect(query_params[:conditions][2]).to eq(Time.now - 21.days)
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.